File: boat-bay-server/src/main/java/com/backbase/oss/boat/bay/web/views/dashboard/models/BoatSpec.java
package com.backbase.oss.boat.bay.web.views.dashboard.models;

import com.backbase.oss.boat.bay.domain.enumeration.Changes;
import com.backbase.oss.boat.bay.service.statistics.BoatStatistics;
import java.time.LocalDateTime;
import lombok.Data;
import org.openapitools.openapidiff.core.model.schema.ChangedEnum;

@Data
public class BoatSpec {

    private long id;

    private String key;
    private String name;
    private String title;
    private String grade;
    private String version;
    private String description;
    private String icon;
    private LocalDateTime createdOn;
    private String createdBy;

    private BoatStatistics statistics;

    private Changes changes;

    private BoatCapability capability;
    private BoatService serviceDefinition;

    private String openApi;
}

File: boat-bay-frontend/src/app/models/boat-spec.ts
import { BoatLintReport } from './boat-lint-report';
import { BoatStatistics } from './boat-dashboard';
import { BoatCapability } from "./boat-capability";
import { BoatService } from "./boat-service";

export enum Changes {
  NOT_APPLICABLE= 'NOT_APPLICABLE', ERROR_COMPARING = 'ERROR_COMPARING', UNCHANGED = "UNCHANGED", COMPATIBLE = "COMPATIBLE", BREAKING = "BREAKING"
}

export interface BoatSpec {

  id: number;
  key: string;
  name: string;
  content?: string;
  openApi?: string;

  createdOn: Date;
  createdBy: String;

  lastLintReport?: BoatLintReport;

  statistics: BoatStatistics;

  version: string;
  title: string;
  description: string;
  icon: string;
  grade: string;

  changes: Changes;

  capability: BoatCapability;
  serviceDefinition: BoatService;
}
